### PR TITLE
fix: fb share method

### PR DESCRIPTION
**Pull request by hakankarakaya**
fix: fb share method

### CODE REVIEW

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 1]
```diff
@@ -62,15 +62,45 @@ private String getMimeType(String filePath) {
     }
   }
 
+  private boolean checkAppInstall(String uri) {
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Let's rename this method to `isAppInstalled` for better readability.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 2]
```diff
@@ -62,15 +62,45 @@ private String getMimeType(String filePath) {
...
+    PackageManager pm = this.reactContext.getPackageManager();
+    try {
+        pm.getApplicationInfo(uri, 0);
+        return true;
+    } catch (PackageManager.NameNotFoundException e) {
+        // Error
+    }
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please use two spaces as tabs. Also correct all tab inconsistencies in this commit.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 3]
```diff
@@ -62,15 +62,45 @@ private String getMimeType(String filePath) {
...
+      ShareDialog shareDialog = new ShareDialog(this.reactContext.getCurrentActivity());
+      ShareLinkContent content = new ShareLinkContent.Builder().setContentUrl(Uri.parse(shareUrl)).build();
+      shareDialog.show(content, ShareDialog.Mode.WEB);
+    } else {
+      String mimeType = this.getMimeType(filePath);
+      if (mimeType.startsWith("image")) {
+          this.shareImageViaFacebook(filePath, promise);
+      } else {
+          this.shareVideoViaFacebook(filePath, promise);
+      }
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please use two spaces as tabs.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 4]
```diff
@@ -80,10 +110,12 @@ private void shareImageViaFacebook(String filePath, final Promise promise) {
 
   private void shareVideoViaFacebook(String filePath, final Promise promise) {
     try {
-      ShareVideo video = new ShareVideo.Builder()
-              .setLocalUrl(this.uriFromFilePath(filePath)).build();
+      ShareVideo video = new ShareVideo.Builder().setLocalUrl(this.uriFromFilePath(filePath)).build();
+
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please remove this extra new line as these three variables are closely related.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 5]
```diff
@@ -187,7 +216,7 @@ public void shareViaInstagramStories(String filePath, String attributionUrl, fin
 
       Activity activity = this.reactContext.getCurrentActivity();
       if (activity.getPackageManager().resolveActivity(intent, 0) != null) {
-        activity.startActivityForResult(intent, 0);
+          activity.startActivityForResult(intent, 0);
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please do not include changes that are not related to this commit's topic (`fix: fb share method`)

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 6]
```diff
@@ -222,15 +281,14 @@ public void shareViaEmail(String filePath, String shareUrl, final Promise promis
       Intent shareIntent = new Intent();
       shareIntent.setAction(Intent.ACTION_SEND);
       shareIntent.setType("message/rfc822");
-      List<ResolveInfo> resolveInfoList =
-              this.reactContext.getPackageManager().queryIntentActivities(shareIntent, 0);
+      List<ResolveInfo> resolveInfoList = this.reactContext.getPackageManager().queryIntentActivities(shareIntent,
+              0);
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please do not include changes that are not related to this commit's topic (`fix: fb share method`)

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 7]
```diff
@@ -222,15 +281,14 @@ public void shareViaEmail(String filePath, String shareUrl, final Promise promis
...
 
       for (ResolveInfo resInfo : resolveInfoList) {
         String packageName = resInfo.activityInfo.packageName;
         String name = resInfo.activityInfo.name;
-        if (packageName.contains("com.android.email") ||
-                packageName.contains("com.google.android.gm") ||
-                packageName.contains("mail")) {
+        if (packageName.contains("com.android.email") || packageName.contains("com.google.android.gm")
+              || packageName.contains("mail")) {
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please do not include changes that are not related to this commit's topic (`fix: fb share method`)

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 8]
```diff
@@ -199,20 +228,50 @@ public void shareViaInstagramStories(String filePath, String attributionUrl, fin
...
+    // ShareDialog.show(this.reactContext.getCurrentActivity(), content);
+    // return;
+    // }
+    //
+    // String mimeType = this.getMimeType(filePath);
+    // if (mimeType.startsWith("image")) {
+    // this.shareImageViaFacebook(filePath, promise);
+    // } else {
+    // this.shareVideoViaFacebook(filePath, promise);
+    // }
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Please remove these lines for clarity, instead of commenting them out. We can easily access this code using the git history and this commit's diff.

**Review by ahmetardal — CHANGES_REQUESTED** [review]
Please use a common grammar for the commit message as we do not use semantic commit messages in this repo. You can check earlier commits for reference:
https://github.com/svhawks/react-native-android-sharer/commits/master

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 2 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: All tab inconsistencies fixed by Android Studio Reformat Code.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 3 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: All tab inconsistencies fixed by Android Studio Reformat Code.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 1 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Renamed.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 4 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Removed.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 3 above)
Comment (ahmetardal) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: I see, but there are tab inconsistencies in the file at the moment. You can check and change this in Android Studio editor preferences. Some if and try catch blocks are indented 2 spaces, some are indented 4 spaces in this commit's changes. Please make them all 2 spaces by hand if Android Studio makes it hard to do.

Screen Shot 2020-01-15 at 23 45 27

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 5 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Fixed some lines with the Android Studio Reformat Code.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 8 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Removed.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 9]
```diff
@@ -62,14 +62,41 @@ private String getMimeType(String filePath) {
...
   }
 
+  private boolean isAppInstalled(String uri) {
+    PackageManager pm = this.reactContext.getPackageManager();
+    try {
+        pm.getApplicationInfo(uri, 0);
+        return true;
+    } catch (PackageManager.NameNotFoundException e) {
+        // Error
+    }
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Tab spacing is still not correct.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 10]
```diff
@@ -62,14 +62,41 @@ private String getMimeType(String filePath) {
...
+      ShareDialog shareDialog = new ShareDialog(this.reactContext.getCurrentActivity());
+      ShareLinkContent content = new ShareLinkContent.Builder().setContentUrl(Uri.parse(shareUrl)).build();
+      shareDialog.show(content, ShareDialog.Mode.WEB);
+    } else {
+      String mimeType = this.getMimeType(filePath);
+      if (mimeType.startsWith("image")) {
+          this.shareImageViaFacebook(filePath, promise);
+      } else {
+          this.shareVideoViaFacebook(filePath, promise);
+      }
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Tab spacing is still not correct.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java [hunk 11]
```diff
@@ -199,19 +227,32 @@ public void shareViaInstagramStories(String filePath, String attributionUrl, fin
...
+        Intent shareIntent = new Intent(Intent.ACTION_SEND);
+        shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+        if (!shareUrl.isEmpty()) {
+            shareIntent.putExtra(Intent.EXTRA_TEXT, shareUrl);
+            shareIntent.setType("text/plain");
+        } else {
+            shareIntent.putExtra(Intent.EXTRA_STREAM, this.uriFromFilePath(filePath));
+            shareIntent.setType(this.getMimeType(filePath));
+        }
```
Comment (ahmetardal) [inline android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Tab spacing is still not correct.

**Review by ahmetardal — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 9 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Fixed.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 10 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Fixed.

File: android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java (hunk 11 above)
Comment (hakankarakaya) [reply android/src/main/java/com/svtek/reactnative/RNAndroidSharerModule.java]: Fixed

**Review by ahmetardal — APPROVED** [review]
